shard/src/ext/latex.py

```diff
@@ -0,0 +1,43 @@
+# Latex command for Architus bot.
+# Takes a string of latex and returns the compiled version as a png
+# Inspired by: https://github.com/chuanshi/slacklatex
+from discord.ext import commands
+from discord import File
+
+from string import Template
+import tempfile
+import subprocess
+import os
+
+
+@commands.command(aliases=['tex'])
+async def latex(ctx, latex):
+    with tempfile.TemporaryDirectory() as work_dir:
+        with open('src/ext/template.tex', 'r',) as f:
+            s = Template(f.read())
+        out_txt = s.substitute(my_text=latex)
+
+        with open(os.path.join(work_dir, 'out.tex'), 'w') as f:
+            f.write(out_txt)
+
+        subprocess.run(['pdflatex', '-halt-on-error', 'out.tex'],
```

Comment:
we need to use `asyncio.subprocess`... 🙂 have fun

Reply:
Wow. That was a fun time.